### PR TITLE
feat: log when delegation is performed or Charge was a tax

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR will add additional logning when
- Message is delegated to a Actor
- Message is sent to a grid area owner.

## References
https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/229

## Checklist
- [x] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [ ] Subsystem test executed (dev_002/dev_003)
- [ ] Is there time to monitor state of the release to Production?
- [x] Reference to the task

### CODE REVIEW

File: source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.EDI.BuildingBlocks.Domain.Models;
+using NodaTime;
+
+namespace Energinet.DataHub.EDI.MasterData.Interfaces.Models;
+
+public record GridAreaOwnerDto(string GridAreaCode, Instant ValidFrom, ActorNumber ActorNumber, int SequenceNumber);
```
Comment (MadsDue) [inline source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs:20]: Is this a GridAreaOwner? 

It should a bit more like a GridAreaOwnership

Comment (MadsDue) [reply source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs:20]: Instant ValidFrom
int SequenceNumber

Not sure why we want to expose those two fields. Aren't they just a concern for the domain model?

File: source/OutgoingMessages.Application/UseCases/DelegateMessage.cs [hunk 2]
```diff
@@ -83,6 +83,16 @@ public async Task<OutgoingMessage> DelegateAsync(
...
+                "Message was delegated from {FromActorNumber} to {ToActorNumber}, with role {ActorRole} based on delegation with sequence number {SequenceNumber}",
+                delegatedByActorNumber.Value,
+                messageDelegation.DelegatedTo.ActorNumber.Value,
+                delegatedByActorRole.Code,
+                messageDelegation.SequenceNumber);
+        }
+
         return messageDelegation is not null
             ? Receiver.Create(messageDelegation.DelegatedTo.ActorNumber, messageDelegation.DelegatedTo.ActorRole)
             : null;
```
Comment (MadsDue) [inline source/OutgoingMessages.Application/UseCases/DelegateMessage.cs:98]: Could we log the receiver instead? 
- Receiver.Create() could potentiel do should else.

```suggestion
var receiver = messageDelegation is not null
            ? Receiver.Create(messageDelegation.DelegatedTo.ActorNumber, messageDelegation.DelegatedTo.ActorRole)
            : null
        if (receiver != null)
        {
            _logger.LogInformation(
                "Message was delegated from {FromActorNumber} to {ToActorNumber}, with role {ActorRole} based on delegation with sequence number {SequenceNumber}",
                delegatedByActorNumber.Value,
                receiver.ActorNumber.Value,
                 receiver.ActorRode.Name,
                messageDelegation.SequenceNumber);
        }

        return receiver;
```

File: source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs (hunk 1 above)
Comment (lasrinnil) [reply source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs:20]: `ValidFrom` is just for now a concern for the domain model. But the `SequenceNumber` is used when logging that the GridAreaOwner ActorNumber is being used. If we had a EventId we could have perhaps chosen this instead.  But this way we log with an identifier which could be used to track the information back to `MarketParticipant` subsystem.

Comment (lasrinnil) [reply source/MasterData.Interfaces/Models/GridAreaOwnerDto.cs:20]: > Is this a GridAreaOwner?
> 
> It should a bit more like a GridAreaOwnership

Whats the difference in this case? In daily talk we use `GridAreaOwner` imo.

File: source/OutgoingMessages.Application/UseCases/DelegateMessage.cs (hunk 2 above)
Comment (lasrinnil) [reply source/OutgoingMessages.Application/UseCases/DelegateMessage.cs]: please see my last commit 👍

**Review by MadsDue — APPROVED** [review]

**Pull request merged**